packages/application-components/src/components/modal-pages/internals/modal-page.tsx

```diff
@@ -125,57 +117,69 @@ const ModalPage = ({
     },
     [onClose, shouldDelayOnClose, TRANSITION_DURATION]
   );
+
+  const [portalContainer, setPortalContainer] = useState<HTMLElement | null>(
+    null
+  );
+
+  useEffect(() => {
+    const container = getParentSelector();
+    if (container) {
+      setPortalContainer(container);
+    }
+  }, []);
```

Comment:
I mentioned the other day that when the URL is refreshed, the order of modal rendering is reversed. This change was needed (the use of local state and useEffect to manage the portals container) to make it work